pkg/arvo/sur/btc-node-hook.hoon

```diff
@@ -0,0 +1,2489 @@
+          ::
+            is-compressed=(unit ?)
+          ::
+            label=(unit @t)
+          ::
+            timestamp=(unit @ud)
+          ::
+            hd-key-path=(unit @t)
+          ::
+            hd-seed-id=(unit @ux)
+          ::
+            hd-master-finger-print=(unit @ux)
+          ::
+            labels=(list [name=@t =purpose])
+        ==
+      ::
+        [%get-balance amount=@t]
+      ::
+        [%get-new-address =address]
+      ::
+        [%get-raw-change-address =address]
+      ::
+        [%get-received-by-address amount=@t]
+      ::
+        [%get-received-by-label amount=@t]
+      ::
+        $:  %get-transaction
+            amount=@t
+          ::
+            fee=(unit @t)
+          ::
+            confirmations=@ud
+          ::
+            blockhash=(unit @ux)
+          ::
+            blockindex=(unit @ud)
+          ::
+            blocktime=(unit @ud)
+          ::
+            txid=@ux
+          ::
+            time=@ud
+          ::
+            time-received=@ud
+          ::
+            =bip125-replaceable
+          ::
+            $=  details
+            %-  list
+            $:  address=(unit address)
+              ::
+                =category
+              ::
+                amount=@t
+              ::
+                label=(unit @t)
+              ::
+                vout=@ud
+              ::
+                fee=(unit @t)
+              ::
+                abandoned=(unit ?)
+            ==
+          ::
+            hex=@ux
+        ==
+      ::
+        [%get-unconfirmed-balance amount=@t]
+      ::
+        $:  %get-wallet-info
+            wallet-name=@t
+          ::
+            wallet-version=@ud
+          ::
+            balance=@t
+          ::
+            unconfirmed-balance=@t
+          ::
+            immature-balance=@t
+          ::
+            tx-count=@ud
+          ::
+            key-pool-oldest=@ud
+          ::
+            key-pool-size=@ud
+          ::
+            key-pool-size-hd-internal=(unit @ud)
+          ::
+            unlocked-until=(unit @ud)
+          ::
+            pay-tx-fee=@t
+          ::
+            hd-seed-id=(unit @ux)
+          ::
+            private-keys-enabled=?
+        ==
+      ::
+        [%import-address ~]
+      ::
+        $:  %import-multi
+            %-  list
+            $:  success=?
+              ::
+                warnings=(unit (list @t))
+              ::
+                errors=(unit [code=@t message=@t])
+        ==  ==
+      ::
+        [%import-privkey ~]
+      ::
+        [%import-pruned-funds ~]
+      ::
+        [%import-pubkey ~]
+      ::
+        [%import-wallet ~]
+      ::
+        [%key-pool-refill ~]
+      ::
+        $:  %list-address-groupings
+            $=  groups
+            %-  list
+            %-  list
+            %-  list
+            $:  address
+             ::
+                amount=@t
+              ::
+                label=(unit @t)
+        ==  =
```

Comment:
Should this be done within this PR?

Reply:
whoops, I completely missed this comment and now I can't see where was pointing at. Was it about `@uc` not parsing segwit addresses? If so I would say "later..."